code/datums/traits/negative.dm

```diff
@@ -614,6 +614,51 @@
 			return
 		SEND_SIGNAL(quirk_holder, COMSIG_ADD_MOOD_EVENT, "wrong_cigs", /datum/mood_event/wrong_brand)
 
+
+/datum/quirk/junkie/drunkard
+	name = "Drunkard"
+	desc = "In space there's no such thing as day drinking."
+	icon = "beer" 
+	value = -2
+	mood_quirk = TRUE
+	gain_text = span_danger("You could really go for a stiff drink right about now.")
+	lose_text = span_notice("You no longer feel dependent on alcohol to function.")
+	medical_record_text = "Patient is known to be dependent on alcohol."
+	reagent_type = /datum/reagent/consumable/ethanol
+	
+/datum/quirk/junkie/drunkard/on_spawn()
+	var/mob/living/carbon/human/H = quirk_holder
+	H.reagents.add_reagent(/datum/reagent/consumable/ethanol, 20)
+	drug_container_type = pick(/obj/item/reagent_containers/food/drinks/beer/light/plastic)
+	. = ..()
+
+/datum/quirk/junkie/drunkard/on_process()
+	var/mob/living/carbon/human/H = quirk_holder
+	if (tick_counter == 60) 
+		var/in_list = FALSE
+		for (var/datum/reagent/entry in H.reagents.addiction_list)
+			if(istype(entry, reagent_type))
+				in_list = TRUE
+				break
+		if(!in_list)
+			H.reagents.addiction_list += reagent_instance
+			reagent_instance.addiction_stage = 0
+			to_chat(quirk_holder, span_danger("You suddenly feel like you need another drink..."))
+		tick_counter = 0
+	else
+		++tick_counter
```

Comment:
rather than rewriting the entire on_process just to change a string, make that string into a variable and modify it for the new quirk

Reply:
A much better way to handle it, thanks!